alacritty/src/event.rs

```diff
@@ -1999,17 +2070,26 @@ impl input::Processor<EventProxy, ActionContext<'_, Notifier, EventProxy>> {
                     WindowEvent::Occluded(occluded) => {
                         *self.ctx.occluded = occluded;
                     },
-                    WindowEvent::DroppedFile(path) => {
-                        let path: String = path.to_string_lossy().into();
-                        self.ctx.paste(&(path + " "), true);
+                    WindowEvent::DragDropped { paths, .. } => {
+                        for path in paths {
+                            let path: String = path.to_string_lossy().into();
+                            self.ctx.paste(&(path + " "), true);
+                        }
                     },
-                    WindowEvent::CursorLeft { .. } => {
-                        self.ctx.mouse.inside_text_area = false;
+                    WindowEvent::PointerLeft { kind, position, .. } => {
+                        if let PointerKind::Touch(finger_id) = kind {
+                            let phase = TouchPhase::Cancelled;
```

Comment:
Why is this called cancelled? A cancelled touch event generally invalidates all previous touch events for the purpose of gesture recognition in the compositor. The concept of the pointer leaving the window and touch sequence being cancelled should be two completely distinctive events…

Reply:
I mean, winit doesn't have a concept of cancel, we just use it in alacritty as touch left means that we cancel our input. When winit gets cancel event, it marks all touch events as `left`, but without `release`. if you release a finger, you get `release -> left`.